layers/sync/sync_error_messages.cpp

```diff
@@ -478,23 +505,94 @@ std::string ErrorMessages::RenderPassStoreOpError(const HazardResult& hazard, co
+}
+
+std::string ErrorMessages::RenderPassFinalLayoutTransitionError(const HazardResult& hazard,
+                                                                const CommandBufferAccessContext& cb_context, vvl::Func command,
+                                                                const std::string& resource_description, VkImageLayout old_layout,
+                                                                VkImageLayout new_layout) const {
+    const char* old_layout_str = string_VkImageLayout(old_layout);
+    const char* new_layout_str = string_VkImageLayout(new_layout);
+
+    AdditionalMessageInfo additional_info;
+    additional_info.properties.Add(kPropertyOldLayout, old_layout_str);
+    additional_info.properties.Add(kPropertyNewLayout, new_layout_str);
+    additional_info.access_action = "performs final image layout transition";
+    return Error(hazard, cb_context, command, resource_description, additional_info);
+}
+
+std::string ErrorMessages::RenderPassFinalLayoutTransitionVsStoreOrResolveError(const HazardResult& hazard,
+                                                                                const CommandBufferAccessContext& cb_context,
+                                                                                vvl::Func command,
+                                                                                const std::string& resource_description,
+                                                                                VkImageLayout old_layout, VkImageLayout new_layout,
+                                                                                uint32_t store_resolve_subpass) const {
+    const char* old_layout_str = string_VkImageLayout(old_layout);
+    const char* new_layout_str = string_VkImageLayout(new_layout);
+
+    AdditionalMessageInfo additional_info;
+    additional_info.properties.Add(kPropertyOldLayout, old_layout_str);
+    additional_info.properties.Add(kPropertyNewLayout, new_layout_str);
+    additional_info.access_action = "performs final image layout transition";
+    additional_info.brief_description_end_text = "during store/resolve operation in subpass ";
+    additional_info.brief_description_end_text += std::to_string(store_resolve_subpass);
+
+    return Error(hazard, cb_context, command, resource_description, additional_info);
+}
+
+std::string ErrorMessages::ImagePipelineBarrierError(const HazardResult& hazard, const CommandBufferAccessContext& cb_context,
+                                                     vvl::Func command, const std::string& resource_description,
+                                                     const SyncImageMemoryBarrier& barrier) const {
+    AdditionalMessageInfo additional_info;
+    additional_info.access_action = "performs image layout transition on the";
+
+    std::stringstream ss;
+    ss << "\nImage barrier: {\n";
+    ss << "  barrier index = " << barrier.barrier_index << ",\n";
+    ss << "  src access = "
```

Comment:
below we use `srcStageMask` (which maps to the API name) so would do

```suggestion
    ss << "  srcAccessMask = "
```

Reply:
Yeah, was aware of this area (explanation below).. This part is something that need to be improved but trying to finish with the remaining messages to be able to remove old helpers, then the plan to do a round of refactorings and more polish.

The current tracking stores exact form of stages (srcStage/dstStage) but accesses are stored in *internal* form. That's why different naming. This will require a bit of refactoring, for now I will update naming further to show it's not a variable name (*source accesses*). The corresponding fields are printed in the form stage(access1|sccess2).